dev/doc/xml-protocol.md

```diff
@@ -304,6 +305,31 @@ Pseudocode for listing all of the goals in order: `rev (flat_map fst background)
 
 -------------------------------
 
+### <a name="command-subgoals">**Subgoals(flags: goal_flags)**</a>
+Similar to [Goal](#command-goal), but with `flags` to control whether to include
+information about `fg`, `bg`, `shelved`, or `given_up` goals. The flags also
+include a `mode`, which is either "full" (return hypotheses and conclusion) or
+"short" (return only conclusion). The "short" mode is useful for speeding up
+goal display when there are many shelved or admitted goals with large proof
+contexts, but the IDE only needs to know their conclusions or how many there
+are.
+```html
+<call val="Subgoals">
+  <goal_flags>
+    <string>${mode}</string>
+    <bool val="${fg}"/>
+    <bool val="${bg}"/>
+    <bool val="${shelved}"/>
+    <bool val="${given_up}"/>
+  </goal_flags>
+</call>
+```
+
+#### Returns
+* The same as [Goal](#command-goal). Goals filtered out by `flags` will be empty
+  lists.
```

Comment:
```suggestion
* The same as [Goal](#command-goal).
```

Seems unnecessary to say that it can return empty lists--what else could the call possibly do?